projects/vdk-core/src/vdk/internal/builtin_plugins/connection/connection_hooks.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from typing import Any
+from typing import cast
+
+from vdk.api.plugin.connection_hook_spec import ConnectionHookSpec
+from vdk.api.plugin.hook_markers import hookimpl
+from vdk.api.plugin.plugin_registry import IPluginRegistry
+from vdk.internal.builtin_plugins.connection.execution_cursor import (
+    ExecuteOperationResult,
+)
+from vdk.internal.builtin_plugins.connection.execution_cursor import ExecutionCursor
+from vdk.internal.core.errors import ErrorMessage
+from vdk.internal.core.errors import PlatformServiceError
+
+
+class DefaultConnectionHookImpl:
+    """
+    The default implementation of execute operation.
+    Generally it should not be overridden.
+    To "override" with new implementation, make a new hook which returns non-None ExecuteOperationResult.
```

Comment:
In such a case, does the default implementation remain evaluated, and only the result is augmented?
Having a "default hook implementation" differs majorly from other hook specs so far, what's the benefit here (pros & cons)? What are the alternatives to be compliant with the already established approaches?

Reply:
This is not a new pattern. It is used in [run_job](https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/api/plugin/core_hook_spec.py#L145) and [run_step](https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/api/plugin/core_hook_spec.py#L171) hooks.